internal/security/grantvmgroupaccess_test.go

```diff
@@ -0,0 +1,289 @@
+// S-1-5-83-1-3166535780-1122986932-343720105-43916321:(OI)(CI)(R,W)
+//
+// File in directory (inherited):
+// S-1-15-3-1024-2268835264-3721307629-241982045-173645152-1490879176-104643441-2915960892-1612460704:(I)(R,W)
+// S-1-5-83-1-3166535780-1122986932-343720105-43916321:(I)(R,W)
+
+func TestGrantVmGroupAccessDefault(t *testing.T) {
+	f, err := ioutil.TempFile("", "gvmgafile")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		f.Close()
+		os.Remove(f.Name())
+	}()
+
+	d, err := ioutil.TempDir("", "gvmgadir")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(d)
+
+	find, err := os.Create(filepath.Join(d, "find.txt"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := GrantVmGroupAccess(f.Name()); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := GrantVmGroupAccess(d); err != nil {
+		t.Fatal(err)
+	}
+
+	verifyVMAccountDACLs(t,
+		f.Name(),
+		[]string{`(R)`},
+	)
+
+	// Two items here:
+	//  - One explicit read only.
+	//  - Other applies to this folder, subfolders and files
+	//      (OI): object inherit
+	//      (CI): container inherit
+	//      (IO): inherit only
+	//      (GR): generic read
+	//
+	// In properties for the directory, advanced security settings, this will
+	// show as a single line "Allow/Virtual Machines/Read/Inherited from none/This folder, subfolder and files
+	verifyVMAccountDACLs(t,
+		d,
+		[]string{`(R)`, `(OI)(CI)(IO)(GR)`},
+	)
+
+	verifyVMAccountDACLs(t,
+		find.Name(),
+		[]string{`(I)(R)`},
+	)
+
+}
+
+func TestGrantVMGroupAccess_File_DesiredPermissions(t *testing.T) {
+	type config struct {
+		name                string
+		desiredAccess       accessMask
+		expectedPermissions []string
+	}
+
+	for _, cfg := range []config{
+		{
+			name:                "Read",
+			desiredAccess:       AccessMaskRead,
+			expectedPermissions: []string{`(R)`},
+		},
+		{
+			name:                "Write",
+			desiredAccess:       AccessMaskWrite,
+			expectedPermissions: []string{`(W,Rc)`},
+		},
+		{
+			name:                "Execute",
+			desiredAccess:       AccessMaskExecute,
+			expectedPermissions: []string{`(Rc,S,X,RA)`},
+		},
+		{
+			name:                "ReadWrite",
+			desiredAccess:       AccessMaskRead | AccessMaskWrite,
+			expectedPermissions: []string{`(R,W)`},
+		},
+		{
+			name:                "ReadExecute",
+			desiredAccess:       AccessMaskRead | AccessMaskExecute,
+			expectedPermissions: []string{`(RX)`},
+		},
+		{
+			name:                "WriteExecute",
+			desiredAccess:       AccessMaskWrite | AccessMaskExecute,
+			expectedPermissions: []string{`(W,Rc,X,RA)`},
+		},
+		{
+			name:                "ReadWriteExecute",
+			desiredAccess:       AccessMaskRead | AccessMaskWrite | AccessMaskExecute,
+			expectedPermissions: []string{`(RX,W)`},
+		},
+		{
+			name:                "All",
+			desiredAccess:       AccessMaskAll,
+			expectedPermissions: []string{`(F)`},
+		},
+	} {
+		t.Run(cfg.name, func(t *testing.T) {
+			dir := t.TempDir()
```

Comment:
nit: you can use testing.T's "TempDir" instead

Reply:
Isn't that what it's doing?

Reply:
I'm guessing this is related to the `ioutil.Tempfile` usage in the test above?

Reply:
whoops, wrong line haha, I was referring to the use of ioutil